File: 101-109_small_problems/easy_8/fizzbuzz.rb
# fizzbuzz
# Write a method that takes two arguments: the first is the starting number, and the second is the ending number. Print out all numbers between the two numbers, except if a number is divisible by 3, print "Fizz", if a number is divisible by 5, print "Buzz", and finally if a number is divisible by 3 and 5, print "FizzBuzz".

# Input: Two integer arguments
# Output: A string
# Mental model: Given two numbers, output the numbers between and including those numbers. If the numbers are divisible by 3, 5, or 15, output Fizz, Buzz, or FizzBuzz as appropriate.

# Example:

# fizzbuzz(1, 15) # -> 1, 2, Fizz, 4, Buzz, Fizz, 7, 8, Fizz, Buzz, 11, Fizz, 13, 14, FizzBuzz

# Data Structure

# Input are two integers. Output is a string.

# Algorithm
# Loop second number times
# Output current number
#   If current number is divisible by 15
#     Output FizzBuzz
#   Else if current number is divisible by 3
#     Output fizz
#   Else if current number is divisible by 5
#     Output buzz
#   Else
#     Output number

# Code

def fizzbuzz(start, finish)
  start.upto(finish) do |number|

   output = case
     when number % 15 == 0 then "FizzBuzz"
     when number % 3 == 0 then "Fizz"
     when number % 5 == 0 then "Buzz"
     else
       number
     end

    print output
    print ", " if number != finish
  end
end

fizzbuzz(1, 15) #=> "1, 2, Fizz, 4, Buzz, Fizz, 7, 8, Fizz, Buzz, 11, Fizz, 13, 14, FizzBuzz"

File: lessons/210_lessons/lesson_3/fizzBuzz.js
// FizzBuzz
// Write a function that iterates over the integers from 1 to 100, inclusive. For multiples of three, log "Fizz" to the console. For multiples of five, log "Buzz". For numbers which are multiples of both three and five, log "FizzBuzz". For all other numbers, log the number.
//
// Your output should look like this:
//
// fizzbuzz();
//
// // console output
// 1
// 2
// Fizz
// 4
// Buzz
// Fizz
// 7
// 8
// Fizz
// Buzz
// 11
// Fizz
// 13
// 14
// FizzBuzz
// 16
// 17
// Fizz
// // ... rest of output omitted for brevity

// Input: None
// Output: Numbers from 1 to 100, inclusive. Fizz if multiple of 3, Buzz if multiple of 5, and FizzBuzz  if multiple of 3 and 5.

// Algorithm
// 1. Loop from 1 to 100.
//   If current number is a multiple of 3 and 5
//     Output 'FizzBuzz'
//   Else if current number is a multiple of 3
//     Output 'Fizz'
//   Else if current number is a multiple of 5
//     Output 'Buzz'
//   Else
//     Output number

function fizzbuzz() {
  for (var i = 1; i <= 100; i++) {
    if (i % 3 === 0 && i % 5 === 0) {
      console.log('FizzBuzz');
    } else if (i % 3 === 0) {
      console.log('Fizz');
    } else if (i % 5 === 0) {
      console.log('Buzz');
    } else {
      console.log(i.toString());
    }
  }
}

fizzbuzz();
